### ISSUE TITLE
raft_test test case failure

### ISSUE THREAD

**Issue by achamayou** [microsoft/CCF#4558]
https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=54312&view=logs&j=f3457e4b-102d-5a4f-f801-dc119e4853a8&t=ae98f324-492e-5969-593b-258ede9c5463

```
9: Test command: /__w/2/s/build/raft_test
9: Test timeout computed to be: 10000000
9: [doctest] doctest version is "2.4.8"
9: [doctest] run with "--help" for options
9: Attempted 48/48 roundtrips
9: ===============================================================================
9: ../src/consensus/aft/test/committable_suffix.cpp:477:
9: TEST CASE:  Multi-term divergence
9: 
9: DEEPEST SUBCASE STACK REACHED (DIFFERENT FROM THE CURRENT ONE):
9: 
9: ../src/consensus/aft/test/committable_suffix.cpp:822: FATAL ERROR: REQUIRE( rA.get_last_idx() == rB.get_last_idx() ) is NOT correct!
9:   values: REQUIRE( 36 == 37 )
9:   logged: Bring other node in-sync
9:           The final state is synced on all nodes
```

**Comment by eddyashton** [comment]
This test has a random factor, and we don't log the seed. I've made a local modification to log the seed (in the hope that if we see this again, it's a consistent repro). Currently running in a loop, but no hits yet.

**Comment by achamayou** [comment]
Is the seed logging done to stdout? ctest seems to supress that unforunately.

**Comment by eddyashton** [comment]
We can log directly to stdout in a doctest-in-ctest run - that's how the `Attempted 48/48 roundtrips` in [the same test](https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=54312&view=logs&j=f3457e4b-102d-5a4f-f801-dc119e4853a8&t=ae98f324-492e-5969-593b-258ede9c5463&l=840) is logged.

**Comment by eddyashton** [comment]
Found a consistently failing seed `1668601152`:

```
$ ./raft_test 
[doctest] doctest version is "2.4.8"
[doctest] run with "--help" for options
Attempted 47/47 roundtrips
===============================================================================
../src/consensus/aft/test/committable_suffix.cpp:477:
TEST CASE:  Multi-term divergence

DEEPEST SUBCASE STACK REACHED (DIFFERENT FROM THE CURRENT ONE):

../src/consensus/aft/test/committable_suffix.cpp:824: FATAL ERROR: REQUIRE( rA.get_last_idx() == rB.get_last_idx() ) is NOT correct!
  values: REQUIRE( 34 == 35 )
  logged: Using seed: 1668601152
          Bring other node in-sync
          The final state is synced on all nodes

===============================================================================
[doctest] test cases:      14 |      13 passed | 1 failed | 0 skipped
[doctest] assertions: 1000761 | 1000760 passed | 1 failed |
[doctest] Status: FAILURE!
(env) edashton@edashton2:/data/src/2.CCF/build$ ./raft_test 
[doctest] doctest version is "2.4.8"
[doctest] run with "--help" for options
Attempted 47/47 roundtrips
===============================================================================
../src/consensus/aft/test/committable_suffix.cpp:477:
TEST CASE:  Multi-term divergence

DEEPEST SUBCASE STACK REACHED (DIFFERENT FROM THE CURRENT ONE):

../src/consensus/aft/test/committable_suffix.cpp:824: FATAL ERROR: REQUIRE( rA.get_last_idx() == rB.get_last_idx() ) is NOT correct!
  values: REQUIRE( 34 == 35 )
  logged: Using seed: 1668601152
          Bring other node in-sync
          The final state is synced on all nodes

===============================================================================
[doctest] test cases:      14 |      13 passed | 1 failed | 0 skipped
[doctest] assertions: 1000761 | 1000760 passed | 1 failed |
[doctest] Status: FAILURE!
```

**Comment by eddyashton** [comment]
Resolved by #4589.
